src/ununpack/agent/checksum.c

```diff
@@ -243,3 +250,39 @@ char *	SumToString	(Cksum *Sum)
   return(Result);
 } /* SumToString() */
 
+int calc_sha256sum(char*filename, char* dst) {
+  gcry_md_hd_t checksumhandler;
+  unsigned char buf[32];
+  unsigned char *tempBuff;
+  memset(buf, '\0', sizeof(buf));
+  FILE *f;
+  if(!(f=fopen(filename, "rb")))
+  {
+    LOG_FATAL("Failed to open file '%s'\n", filename);
+    return(1);
+  }
+  gcry_md_open(&checksumhandler, GCRY_MD_SHA256, 0);
+  if (! checksumhandler)
```

Comment:
int gcry_md_is_enabled (gcry_md_hd_t h, int algo) seems to be a better check

Reply:
Done, please check.